standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java

```diff
@@ -107,6 +107,17 @@ public static Date convertStringToDate(String date) {
     return java.sql.Date.valueOf(val);
   }
 
+  /**
+   * Converts the string format date without a time-zone to
+   * a time-zone based string format date
+   * @param date the date without a time-zone
+   * @return time-zone based string format date
+   */
+  public static String normalizeDate(String date) {
+    Date val = MetaStoreUtils.convertStringToDate(date);
```

Comment:
minor, up to you:
````
return convertDateToString(convertStringToDate(date));
````
also we could reuse the new method in `BaseSemanticAnalyzer#normalizeDateCol`

Reply:
done